### PR TITLE
Cleanup csproj files

### PR DESCRIPTION
**Pull request by arturcic**
In this PR I have cleaned up the csproj using the defaults of the SDK project. used [this](https://www.michaeltaylorp3.net/migrating-to-sdk-project-format/) as guideline

### CODE REVIEW

File: src/GitVersionCore/GitVersionCore.csproj [hunk 1]
```diff
@@ -21,36 +21,11 @@
...
-    <GenerateAssemblyFileVersionAttribute Condition="'$(GenerateAssemblyFileVersionAttribute)' == ''">false</GenerateAssemblyFileVersionAttribute>
-    <GenerateAssemblyInformationalVersionAttribute Condition="'$(GenerateAssemblyInformationalVersionAttribute)' == ''">false</GenerateAssemblyInformationalVersionAttribute>
-    <GenerateAssemblyProductAttribute Condition="'$(GenerateAssemblyProductAttribute)' == ''">false</GenerateAssemblyProductAttribute>
-    <GenerateAssemblyTitleAttribute Condition="'$(GenerateAssemblyTitleAttribute)' == ''">false</GenerateAssemblyTitleAttribute>
-    <GenerateAssemblyVersionAttribute Condition="'$(GenerateAssemblyVersionAttribute)' == ''">false</GenerateAssemblyVersionAttribute>
-    <GenerateNeutralResourcesLanguageAttribute Condition="'$(GenerateNeutralResourcesLanguageAttribute)' == ''">true</GenerateNeutralResourcesLanguageAttribute>
-  </PropertyGroup>
 
-  <PropertyGroup Condition="'$(TargetFramework)' == 'net40'">
-    <DefineConstants>TRACE;NET40;NETDESKTOP</DefineConstants>
```
Comment (arturcic) [inline src/GitVersionCore/GitVersionCore.csproj]: removed TRACE;NET40; constants as they are not use (TRACE is included by sdk)

File: src/GitVersionExe/GitVersionExe.csproj [hunk 2]
```diff
@@ -1,53 +1,23 @@
...
-
-    <DocumentationFile>bin\Release\GitVersion.xml</DocumentationFile>
-    <NoWarn>1591</NoWarn>
     <PlatformTarget>AnyCPU</PlatformTarget>
+    <DocumentationFile>bin\$(Configuration)\GitVersion.xml</DocumentationFile>
+    <NoWarn>1591</NoWarn>
   </PropertyGroup>
 
   <PropertyGroup Condition="'$(TargetFramework)' == 'net40'">
-    <DefineConstants>NET40;NETDESKTOP</DefineConstants>
```
Comment (arturcic) [inline src/GitVersionExe/GitVersionExe.csproj]: removed NET40; constants as it is not used

**Comment by arturcic** [comment]
I have updated the PR with 2 more changes

File: src/GitVersionTask.Tests/GitVersionTask.Tests.csproj [hunk 3]
```diff
@@ -55,36 +23,18 @@
...
-    <Reference Include="Microsoft.CSharp" />
-    <Reference Include="System" />
-    <Reference Include="System.ComponentModel.Composition" />
-    <Reference Include="System.Core" />
-    <Reference Include="System.Runtime.Serialization" />
-    <Reference Include="System.Xml.Linq" />
-    <Reference Include="System.Data.DataSetExtensions" />
-    <Reference Include="System.Data" />
-    <Reference Include="System.Xml" />
+    <!--See https://github.com/Microsoft/msbuild/issues/3671 for reason for the ExcludeAssets against msbuild packagerefernces -->
```
Comment (dazinator) [inline src/GitVersionTask.Tests/GitVersionTask.Tests.csproj]: If you follow the link above, I think there should be an excludeasset present here? Perhaps we should add it..

Comment (arturcic) [reply src/GitVersionTask.Tests/GitVersionTask.Tests.csproj]: you mean based on [this ](https://github.com/Microsoft/msbuild/issues/3671#issuecomment-418863446)comment?

Comment (arturcic) [reply src/GitVersionTask.Tests/GitVersionTask.Tests.csproj]: I will add that

Comment (arturcic) [reply src/GitVersionTask.Tests/GitVersionTask.Tests.csproj]: let's wait for the build to finish and if there is no other changes you think I should do we can merge it

Comment (dazinator) [reply src/GitVersionTask.Tests/GitVersionTask.Tests.csproj]: :-( looks like build failure

Comment (dazinator) [reply src/GitVersionTask.Tests/GitVersionTask.Tests.csproj]: I think it looks like for the tests project, they still need to be output, as it probably doesn't know how to resolve them, where as msbuild in the wild does..

Comment (dazinator) [reply src/GitVersionTask.Tests/GitVersionTask.Tests.csproj]: :-) probs best to add it back for the tests project then, and delete that confusing comment? I assume the comment must only be relevent for GitVersionTask.csproj!

Comment (arturcic) [reply src/GitVersionTask.Tests/GitVersionTask.Tests.csproj]: I'll do that

Comment (arturcic) [reply src/GitVersionTask.Tests/GitVersionTask.Tests.csproj]: removed the comment, reverted the change and squashed to have one commit

**Comment by dazinator** [comment]
Nice one ;-)

**Comment by arturcic** [comment]
Thanks

**Pull request merged**